example/cpp-example/cronos.cc

```diff
@@ -493,6 +521,111 @@ void test_erc20_total_supply() {
+  w->add_address(receiverAddress);                   // to
+  w->add_uint(tokenid);                              // tokenId
+  Vec<uint8_t> data = w->encode("safeTransferFrom"); // encoded
+  std::string hexstring = bytes_to_hexstring((char *)data.data(), data.size());
+  char hdpath[100];
+  int cointype = 60;
+  int chainid = mychainid; // defined in cronos-devnet.yaml
+  snprintf(hdpath, sizeof(hdpath), "m/44'/%d'/0'/0/0", cointype);
+  Box<PrivateKey> privatekey = mywallet->get_key(hdpath);
+
+  EthTxInfoRaw eth_tx_info = new_eth_tx_info();
+  eth_tx_info.to_address = mycontract;
+  eth_tx_info.nonce = thisNonce;
+  eth_tx_info.amount = "0";
+  eth_tx_info.amount_unit = EthAmount::EthDecimal;
+  eth_tx_info.data = data;
+  eth_tx_info.gas_limit = "219400";
+  eth_tx_info.gas_price = "100000000";
+  eth_tx_info.gas_price_unit = EthAmount::WeiDecimal;
+
+  Vec<uint8_t> signedtx =
+      build_eth_signed_tx(eth_tx_info, chainid, true, *privatekey);
+  String status =
+      broadcast_eth_signed_raw_tx(signedtx, mycronosrpc, 1000).status;
+  cout << "status: " << status << endl;
+}
+
+void test_dynamic_api_call() {
+
+  std::ifstream t("../../common/src/contract/erc721-abi.json");
+  std::stringstream buffer;
+  buffer << t.rdbuf();
+  std::string json = buffer.str();
+
+  String mymnemonics = getEnv("MYMNEMONICS");
+  String mycronosrpc = getEnv("MYCRONOSRPC");
+  String mycontract = getEnv("MYCONTRACT721");
+
+  Box<EthContract> mycontractcall = new_eth_contract(json);
+
+  std::string tokenid;
+  std::cout << "Enter tokenid: ";
+  std::cin >> tokenid;
+
+  mycontractcall->add_uint(tokenid);                     // tokenId
+  Vec<uint8_t> data = mycontractcall->encode("ownerOf"); // encoded
+  std::string response =
+      mycontractcall->call(mycronosrpc, mycontract, "ownerOf").c_str();
+  std::cout << "response: " << response << endl;
+}
+
+void test_dynamic_api_recursive_arg() {
+  std::ifstream t("../../common/src/contract/erc721-abi.json");
+  std::stringstream buffer;
+  buffer << t.rdbuf();
+  std::string json = buffer.str();
+  Box<EthContract> mycontractcall = new_eth_contract(json);
+
+  mycontractcall->begin_tuple();
+  mycontractcall->add_uint("20");
+  mycontractcall->add_uint("30");
+  Box<EthAbiTokenWrapper> datatuple = mycontractcall->commit_tuple();
+  mycontractcall->begin_array();
+  mycontractcall->add_uint("200");
+  mycontractcall->add_uint("300");
+  Box<EthAbiTokenWrapper> dataarray = mycontractcall->commit_array();
+
+  mycontractcall->begin_fixed_array();
+  mycontractcall->add_uint("2000");
+  mycontractcall->add_uint("3000");
+  Box<EthAbiTokenWrapper> datafixedarray = mycontractcall->commit_fixed_array();
+
+  mycontractcall->begin_tuple();
+  mycontractcall->add_uint("1");
+  mycontractcall->add_wrapper(datatuple);
+  mycontractcall->add_wrapper(dataarray);
+  mycontractcall->add_wrapper(datafixedarray);
+  mycontractcall->add_uint("2");
+  Box<EthAbiTokenWrapper> finaltuple = mycontractcall->commit_tuple();
```

Comment:
I find it a little bit hard to follow what the end result type is expected here, so it may be good to write a one-line comment in the Solidity notation to show what's being constructed

Reply:
ok, i'll add more comment